CleverPush/Source/CPAppBannerViewController.m

```diff
@@ -13,14 +13,13 @@ - (void)viewDidLoad {
         return;
     }
 
-    if ([CPAppBannerModuleInstance getCurrentVoucherCodePlaceholder] != nil && [CPAppBannerModuleInstance getCurrentVoucherCodePlaceholder].count > 0) {
-        for (id key in [CPAppBannerModuleInstance getCurrentVoucherCodePlaceholder]) {
+    if ([CPAppBannerModuleInstance getCurrentVoucherCodePlaceholder] != nil) {
+        [[CPAppBannerModuleInstance getCurrentVoucherCodePlaceholder] enumerateKeysAndObjectsUsingBlock:^(id key, id obj, BOOL *stop) {
```

Comment:
does this work? cant you remove below if condition and do
enumerateKeysAndObjectsUsingBlock:^(id self.data.id ...
and what is "id" in this language?

Reply:
Kevin, We need to add a condition for that.
For the filter or map, We need to use the function like above.
I have tested. ID means the datatype.

Reply:
this sounds like it is still looping and not using indexing principle?
okay do you need to use id? because you know that the key is a string, cant you use string there?

Reply:
Done.

Reply:
https://ibb.co/kcZxTGz

Reply:
okay this solves the id stuff(btw whenevery possible use strict types and no dynamic types)
but still this looks like its looping(looks like c# TryGetFirstValue/TryGetNextValue)
it should be possible without loop